### DISCUSSION TITLE
Syncing Multiple Trilium Instances with Different Docker Containers

### DISCUSSION THREAD

**Discussion by NpWas** [TriliumNext/Trilium#4486] [General]
Title: Syncing Multiple Trilium Instances with Different Docker Containers

Hello Trilium Community,

I am running two Trilium instances via Docker on an Ubuntu server (trilium_personal on port 8080 and trilium_work on port 8081), I can access them in my web browser on all computers on my LAN and over VPN.  

I have a Windows 11 machine currently syncing with trilium_personal.   I want to open a second Trilium window on the same Windows machine, that syncs with the trilium_work container.

Challenges:

-If there is documentation addressing this please feel free to point me there, I have not been able to find the answer there so far. I would love to RTFM if thats where the answer is. 

-Attempted to create two separate shortcuts for trilium.exe,  by appending --server=http://address:port to each, as suggested by ChatGPT, but this did not work. No matter what short cut is used, trilium_personal opens by default. 

-I have seen trilium workspaces recommended in some threads, but this does not fit the need.

any assistance is appreciated

**Comment by zerebos** [comment]
> as suggested by ChatGPT

Generally not the best idea...

Trilium by default tries to lock to a single instance of the app running, if you want to have multiple running you have to launch the second one with a different port and data directory using environment variables. See more info here: https://github.com/zadam/trilium/issues/3788#issuecomment-1491016049

**Reply by NpWas** [reply]
trilium_personal is using port 8080 and trilium_work is using port 8081. I created 2 bat files that allow me to change the environment variable and launch trilium.exe. This functions more like toggling between the variables instead of being able to access both data directory's simultaneously in different sessions. Is it possible to have both running simultaneously?

**Comment by zadam** [comment]
So, it appears that you have 2 completely separate databases on the server and just want to have the same on Windows (with 2 syncs for each process on the server).

Basically, you need to configure both different port and different data directory using environment variables. You already have the personal running on the default port and targeting default directory.

What might not be immediately obvious is that Trilium needs a local port on windows as well. By default it runs on 37740 (your personal). So you give it another one - e.g. 37741, give it a different data directory path, and during the setup, point it to your server:8081.
